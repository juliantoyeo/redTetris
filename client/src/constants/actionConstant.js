export const ACTIONS = {
	ALERT_POP: 'ALERT_POP'
}

export const PLAYER_ACTIONS = {
	CREATE_PLAYER: 'CREATE_PLAYER',
	UPDATE_PLAYER: 'UPDATE_PLAYER'
}

export const ROOM_ACTIONS = {
	ADD_ROOM: 'ADD_ROOM',
	DELETE_ROOM: 'DELETE_ROOM',
	UPDATE_ROOM: 'UPDATE_ROOM'
}