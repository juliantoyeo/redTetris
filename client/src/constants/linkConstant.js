export const LINKS = {
	HOME: '/',
	GAME: '/game'
}