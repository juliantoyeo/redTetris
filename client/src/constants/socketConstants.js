export const SOCKET_RES = {
	PLAYER_CREATED: 'PLAYER_CREATED',
	PLAYER_NAME_EXIST: 'PLAYER_NAME_EXIST',
	PLAYER_NOT_EXIST: 'PLAYER_NOT_EXIST',
	ROOM_CREATED: 'ROOM_CREATED',
	ROOM_NAME_EXIST: 'ROOM_NAME_EXIST',
	ROOM_DOESNT_EXIST: 'ROOM_DOESNT_EXIST',
	ROOM_UPDATED: 'ROOM_UPDATED',
	ROOM_DELETED: 'ROOM_DELETED'
}

export const SOCKET_EVENTS = {
	START_GAME: 'START_GAME',
	RESTART_GAME: 'RESTART_GAME',
	CREATE_PLAYER: 'CREATE_PLAYER',
	UPDATE_PLAYER: 'UPDATE_PLAYER',
	FECTH_ALL_ROOM: 'FECTH_ALL_ROOM',
	CREATE_ROOM: 'CREATE_ROOM',
	UPDATE_ROOM: 'UPDATE_ROOM',
	DELETE_ROOM: 'DELETE_ROOM',
	UPDATE_BOARD: 'UPDATE_BOARD',
	GET_NEW_STACK: 'GET_NEW_STACK',
	UPDATE_STACK_INDEX: 'UPDATE_STACK_INDEX',
	ADD_BLOCKING_ROW: 'ADD_BLOCKING_ROW',
	SET_PLAYER_GAME_OVER: 'SET_PLAYER_GAME_OVER',
	GAME_IS_OVER: 'GAME_IS_OVER'
}
