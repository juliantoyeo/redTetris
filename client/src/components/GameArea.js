import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import Board from './Board';
import Button from './subComponents/Button';
import { CELL_SIZE, BOARD_SIZE } from '../constants/gameConstant';

const mainContainerStyle = (props) => {
	return ({
		boxSizing: 'border-box',
		width: props.numberOfPlayer > 3 ? '20%' : '30%',
		maxWidth: '50vw',
		display: 'flex',
		justifyContent: 'space-evenly',
		margin: '1vw',
		fontSize: props.numberOfPlayer > 3 ? '0.8vw' : '1vw',
		border: '1px solid #333'
	});
}

const gameOverStyle = (props) => {
	return ({
		position: 'absolute',
		backgroundColor: 'rgba(50, 50, 50, 0.8)',
		width: 'inherit',
		height: `${props.height}px`,
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',
		color: 'red',
		fontSize: '5em'
	})
}

const previewContainerStyle = (props) => {
	return ({
		display: 'flex',
		alignItems: 'center',
		alignSelf: 'center',
		borderRadius: '1vw',
		width: '80%',
		overflow: 'hidden',
		height: `${props.cellSize * 3}vw`
	});
}

const styles = {
	boardContainer: {
		width: '65%'
	},
	sideContainer: {
		color: 'white',
		width: '35%',
		display: 'flex',
		flexDirection: 'column',
		justifyContent: 'space-around',
		textAlign: 'center',
		padding: '0.5vw'
	},
	infoContainer: {
		display: 'flex',
		flexDirection: 'column',
		justifyContent: 'space-evenly',
		height: '40%'
	}
}

const GameArea = (props) => {
	const { player, nextPiece, startGame, numberOfPlayer } = props;
	const [parentHeight, setParentHeight] = useState(null);
	let cellSize = CELL_SIZE;

	if (numberOfPlayer > 3)
		cellSize = cellSize - 0.5;

	useEffect(() => {
		if (document.getElementById('gameArea')) {
			setParentHeight(document.getElementById('gameArea').clientHeight);
		}
	}, [document.getElementById('gameArea')]);

	return (
		<div style={mainContainerStyle({ numberOfPlayer: numberOfPlayer })} id={'gameArea'}>
			{player.gameOver && 
				<div style={gameOverStyle({ height: parentHeight })}>
					<span>GAME OVER</span>
				</div>
			}
			<div style={{ width: `${cellSize * BOARD_SIZE.WIDTH}vw` }}>
				{player.board && <Board board={player.board} cellSize={cellSize} numberOfPlayer={numberOfPlayer} />}
			</div>
			<div style={styles.sideContainer}>
				<span className={''}>Next Piece</span>
				<div style={previewContainerStyle({ cellSize: cellSize })}>
					<Board board={nextPiece} cellSize={cellSize} numberOfPlayer={numberOfPlayer} mini={true} />
				</div>
				<div style={styles.infoContainer}>
					<span>Player Name : {player.name}</span>
					<span>Score : {player.gameStatus.score}</span>
					<span>Rows : {player.gameStatus.rows}</span>
					<span>Level : {player.gameStatus.level}</span>
					{player.gameOver && <span>GAME OVER!!</span>}
				</div>
				<Button onClick={startGame} type={'button'} text={player.gameOver ? 'Start Game' : 'Quit Game'} />
			</div>
		</div>
	)
}

GameArea.propTypes = {
	player: PropTypes.object,
	board: PropTypes.array,
	nextPiece: PropTypes.array,
	gameOver: PropTypes.bool,
	gameStatus: PropTypes.object,
	startGame: PropTypes.func,
	numberOfPlayer: PropTypes.number
};

export default GameArea;