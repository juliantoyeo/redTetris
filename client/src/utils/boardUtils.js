import { BOARD_SIZE } from '../constants/gameConstant'
import _ from 'lodash'

export const createBoard = () => {
	return (
		Array.from(new Array(BOARD_SIZE.HEIGHT), () => 
			new Array(BOARD_SIZE.WIDTH).fill('0')
		)
		// TEST_BOARD.I_SPIN
	)
}

export const checkCollision = (piece, boardWithLandedPiece, { x: moveX, y: moveY }) => {
	const newBoard = _.cloneDeep(boardWithLandedPiece)
	// console.log("boardWithLandedPiece", boardWithLandedPiece)
	// console.log(piece)
	for (let y = 0; y < piece.shape.length; y += 1) {
		for (let x = 0; x < piece.shape[y].length; x += 1) {
			const moveToY = y + piece.pos.y + moveY
			const moveToX = x + piece.pos.x + moveX
			if (piece.shape[y][x] !== '0') {
				// if (!newBoard[moveToY] || !newBoard[moveToY][moveToX] || newBoard[moveToY][moveToX][1] !== 'clear') {
				if (!newBoard[moveToY] || !newBoard[moveToY][moveToX] || newBoard[moveToY][moveToX] !== '0') {
					// console.log(piece.shape)
					// console.log("collision happen on x : ", x, " y : ", y, " moveToX : ", moveToX, " moveToY : ", moveToY)
					// colPos = { x: x, y: y}
					// return { collided: true, colPos: { x: x, y: y}}
					return true
				}
			}
		}
	}
}

export const getLandPosition = (piece, boardWithLandedPiece) => {
	// console.log("boardWithLandedPiece", boardWithLandedPiece)
	// console.log(piece)
	// let moveY = BOARD_SIZE.HEIGHT - 1
	let moveY = piece.pos.y
	if (piece.pos.y <= 0)
		moveY = 0
	// console.log("checked moveY : ", moveY, " y : ", moveY - piece.pos.y)
	while (!checkCollision(piece, boardWithLandedPiece, { x: 0, y: moveY - piece.pos.y}))
	{
		
		// if (moveY === BOARD_SIZE.HEIGHT - 1)
		// 	return { x: piece.pos.x, y: moveY - 1}
		moveY++
		// console.log("checked moveY : ", moveY, " y : ", moveY - piece.pos.y)
	}
	moveY -= 1
	if (moveY < 0 && piece.type !== 'I') moveY = 0
	// const landPos = { x: piece.pos.x, y: moveY}
	// console.log("landPos : ", landPos)
	return { x: piece.pos.x, y: moveY}
}
