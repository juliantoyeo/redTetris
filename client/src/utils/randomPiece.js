import { PIECES } from '../constants/gameConstant';

export const randomPiece = () => {
	const types = 'IJLOSZT';
	const randType = types[Math.floor(Math.random() * types.length)];
	return [PIECES[randType].shape[0], randType];
}
