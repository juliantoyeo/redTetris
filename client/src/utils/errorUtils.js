import { SOCKET_RES } from '../constants/socketConstants'

export const errorAlert = (error) => {
	switch (error) {
		case SOCKET_RES.PLAYER_NAME_EXIST:
			alert('Name exist, please try a different name');
			break;
		case SOCKET_RES.ROOM_NAME_EXIST:
			alert('Room name exist, please try a different name');
			break;
		case SOCKET_RES.ROOM_DOESNT_EXIST:
			alert('Room does not exist, please try again');
			break;
		default:
			return;
	}
}