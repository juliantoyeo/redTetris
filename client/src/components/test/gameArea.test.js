import React from 'react';
import { shallow } from 'enzyme';

import GameArea from '../GameArea';
import { createBoard } from '../../utils/boardUtils';
import { PIECES } from '../../constants/gameConstant';

const fakePlayer = {
	name: 'fakePlayer',
	gameStatus: {
		score: 0,
		rows: 0,
		level: 1
	},
	gameOver: false,
	board: createBoard()
}

const startGame = jest.fn();

describe('Test GameArea component', () => {
	it('Should render properly', () => {
		const wrapper = shallow(
			<GameArea
				player={fakePlayer}
				nextPiece={PIECES['I'].shape}
				startGame={startGame}
				numberOfPlayer={1}
			/>);
		expect(wrapper).toMatchSnapshot();
	});
	it('Should render game over', () => {
		fakePlayer.gameOver = true;
		const wrapper = shallow(
			<GameArea
				player={fakePlayer}
				nextPiece={PIECES['I'].shape}
				startGame={startGame}
				numberOfPlayer={1}
			/>);
		expect(wrapper).toMatchSnapshot();
	});
	it('Should render properly for 4 player', () => {
		const wrapper = shallow(
			<GameArea
				player={fakePlayer}
				nextPiece={PIECES['I'].shape}
				startGame={startGame}
				numberOfPlayer={4}
			/>);
		expect(wrapper).toMatchSnapshot();
	});
});