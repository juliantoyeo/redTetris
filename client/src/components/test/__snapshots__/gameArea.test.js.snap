// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GameArea component Should render game over 1`] = `
<div
  id="gameArea"
  style={
    Object {
      "border": "1px solid #333",
      "boxSizing": "border-box",
      "display": "flex",
      "fontSize": "1vw",
      "justifyContent": "space-evenly",
      "margin": "1vw",
      "maxWidth": "50vw",
      "position": "relative",
      "width": "30%",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(50, 50, 50, 0.8)",
        "color": "red",
        "display": "flex",
        "fontSize": "5em",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <span>
      GAME OVER
    </span>
  </div>
  <div
    style={
      Object {
        "width": "15vw",
      }
    }
  >
    <Board
      board={
        Array [
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
        ]
      }
      cellSize={1.5}
      numberOfPlayer={1}
    />
  </div>
  <div
    style={
      Object {
        "color": "white",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "padding": "0.5vw",
        "textAlign": "center",
        "width": "35%",
      }
    }
  >
    <span
      className=""
    >
      Next Piece
    </span>
    <div
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderRadius": "1vw",
          "display": "flex",
          "height": "4.5vw",
          "overflow": "hidden",
          "width": "80%",
        }
      }
    >
      <Board
        board={
          Array [
            Array [
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
            ],
            Array [
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
            ],
            Array [
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
            ],
            Array [
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
            ],
          ]
        }
        cellSize={1.5}
        mini={true}
        numberOfPlayer={1}
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "height": "40%",
          "justifyContent": "space-evenly",
        }
      }
    >
      <span>
        Player Name : 
        fakePlayer
      </span>
      <span>
        Score : 
        0
      </span>
      <span>
        Rows : 
        0
      </span>
      <span>
        Level : 
        1
      </span>
    </div>
    <Button
      text="Quit Game"
      type="button"
    />
  </div>
</div>
`;

exports[`Test GameArea component Should render properly 1`] = `
<div
  id="gameArea"
  style={
    Object {
      "border": "1px solid #333",
      "boxSizing": "border-box",
      "display": "flex",
      "fontSize": "1vw",
      "justifyContent": "space-evenly",
      "margin": "1vw",
      "maxWidth": "50vw",
      "position": "relative",
      "width": "30%",
    }
  }
>
  <div
    style={
      Object {
        "width": "15vw",
      }
    }
  >
    <Board
      board={
        Array [
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
        ]
      }
      cellSize={1.5}
      numberOfPlayer={1}
    />
  </div>
  <div
    style={
      Object {
        "color": "white",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "padding": "0.5vw",
        "textAlign": "center",
        "width": "35%",
      }
    }
  >
    <span
      className=""
    >
      Next Piece
    </span>
    <div
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderRadius": "1vw",
          "display": "flex",
          "height": "4.5vw",
          "overflow": "hidden",
          "width": "80%",
        }
      }
    >
      <Board
        board={
          Array [
            Array [
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
            ],
            Array [
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
            ],
            Array [
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
            ],
            Array [
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
            ],
          ]
        }
        cellSize={1.5}
        mini={true}
        numberOfPlayer={1}
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "height": "40%",
          "justifyContent": "space-evenly",
        }
      }
    >
      <span>
        Player Name : 
        fakePlayer
      </span>
      <span>
        Score : 
        0
      </span>
      <span>
        Rows : 
        0
      </span>
      <span>
        Level : 
        1
      </span>
    </div>
    <Button
      text="Quit Game"
      type="button"
    />
  </div>
</div>
`;

exports[`Test GameArea component Should render properly for 4 player 1`] = `
<div
  id="gameArea"
  style={
    Object {
      "border": "1px solid #333",
      "boxSizing": "border-box",
      "display": "flex",
      "fontSize": "0.8vw",
      "justifyContent": "space-evenly",
      "margin": "1vw",
      "maxWidth": "50vw",
      "position": "relative",
      "width": "20%",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(50, 50, 50, 0.8)",
        "color": "red",
        "display": "flex",
        "fontSize": "5em",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <span>
      GAME OVER
    </span>
  </div>
  <div
    style={
      Object {
        "width": "10vw",
      }
    }
  >
    <Board
      board={
        Array [
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
          Array [
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
            "0",
          ],
        ]
      }
      cellSize={1}
      numberOfPlayer={4}
    />
  </div>
  <div
    style={
      Object {
        "color": "white",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "padding": "0.5vw",
        "textAlign": "center",
        "width": "35%",
      }
    }
  >
    <span
      className=""
    >
      Next Piece
    </span>
    <div
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderRadius": "1vw",
          "display": "flex",
          "height": "3vw",
          "overflow": "hidden",
          "width": "80%",
        }
      }
    >
      <Board
        board={
          Array [
            Array [
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
            ],
            Array [
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
              Array [
                "0",
                "0",
                "I",
                "0",
              ],
            ],
            Array [
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                "0",
                "0",
                "0",
                "0",
              ],
            ],
            Array [
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
              Array [
                "0",
                "I",
                "0",
                "0",
              ],
            ],
          ]
        }
        cellSize={1}
        mini={true}
        numberOfPlayer={4}
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "height": "40%",
          "justifyContent": "space-evenly",
        }
      }
    >
      <span>
        Player Name : 
        fakePlayer
      </span>
      <span>
        Score : 
        0
      </span>
      <span>
        Rows : 
        0
      </span>
      <span>
        Level : 
        1
      </span>
    </div>
    <Button
      text="Quit Game"
      type="button"
    />
  </div>
</div>
`;
